test/exercise/fp2/solution.rb

```diff
@@ -1,23 +1,37 @@
 module Exercise
   module Fp2
     class MyArray < Array
-      # Использовать стандартные функции массива для решения задач нельзя.
-      # Использовать свои написанные функции для реализации следующих - можно.
-
-      # Написать свою функцию my_each
       def my_each
+        for element in self
```

Comment:
`for` loop is considered a bad practice because it Ruby it does not create a scope: https://ruby-doc.org/core-2.7.0/doc/syntax/control_expressions_rdoc.html#label-for+Loop

Reply:
Also, try to implement iterator without using of a loop 🙂

Reply:
done